File: app/src/test/java/ru/dimagor555/stocks/data/model/stock/StockTest.java
package ru.dimagor555.stocks.data.model.stock;

import org.junit.Test;

import static junit.framework.TestCase.assertFalse;
import static junit.framework.TestCase.assertTrue;

public class StockTest {
    @Test
    public void toggleFavouriteTest() {
        Stock stock = new Stock("test");

        assertFalse(stock.isFavourite());

        stock.toggleFavourite();

        assertTrue(stock.isFavourite());

        stock.toggleFavourite();

        assertFalse(stock.isFavourite());
    }
}

File: app/src/test/java/ru/dimagor555/stocks/data/model/stock/StockTest.kt
package ru.dimagor555.stocks.data.model.stock

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import ru.dimagor555.stocks.data.model.stock.entity.Stock

class StockTest {
    @Test
    fun `toggle favourite test`() {
        val stock = Stock("test")

        assertFalse(stock.isFavourite)
        stock.toggleFavourite()

        assertTrue(stock.isFavourite)
        stock.toggleFavourite()

        assertFalse(stock.isFavourite)
    }
}